pkg/util/channel/channel.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2022 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package channel
+
+import (
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+const (
+	// notifyTimeout is the timeout for failing to publish an event to the channel.
+	notifyTimeout = time.Second
+)
+
+type eventHandler func(string)
+
+type Subscriber interface {
+	// Subscribe subscribes an eventHandler which will be called when there is an event sent to the channel.
+	// It's not thread-safe and it's supposed to be called serially before first event is published.
+	// The eventHandler is supposed to execute quickly and not perform blocking operation. Blocking operation should be
+	// deferred to a routine that is triggered by the eventHandler.
+	Subscribe(h eventHandler)
+}
+
+type Notifier interface {
+	// Notify sends an event to the channel.
+	Notify(string) bool
+}
+
+// Channel is different from the Go channel which dispatches every event to only single consumer regardless of the
+// number of consumers. Instead, it dispatches every event to all consumers by calling the eventHandlers they have
+// registered.
+type Channel struct {
```

Comment:
Should we use a different name to distinguish from Golang channel? Like SubscriptionChannel.

Reply:
does SubscribableChannel make sense to you?

Reply:
Works for me.